internal/testrunner/runners/system/tester.go

```diff
@@ -1987,10 +1995,78 @@ func (r *tester) checkTransforms(ctx context.Context, config *testConfig, pkgMan
 		if err != nil {
 			return fmt.Errorf("creating fields validator for data stream failed (path: %s): %w", transformRootPath, err)
 		}
-		if errs := validateFields(transformDocs, fieldsValidator); len(errs) > 0 {
-			return testrunner.ErrTestCaseFailed{
-				Reason:  fmt.Sprintf("errors found in documents of preview for transform %s for data stream %s", transformId, dataStream),
-				Details: errs.Error(),
+		if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == fieldsMethod {
+			if errs := validateFields(transformDocs, fieldsValidator); len(errs) > 0 {
+				return testrunner.ErrTestCaseFailed{
+					Reason:  fmt.Sprintf("errors found in documents of preview for transform %s for data stream %s", transformId, dataStream),
+					Details: errs.Error(),
+				}
+			}
+		}
+
+		if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == mappingsMethod {
+			destIndexTransform := transform.Definition.Dest.Index
+			// Index Template format is: "<type>-<package>.<transform>-template"
+			// For instance: "logs-ti_anomali.latest_intelligence-template"
+			indexTemplateTransform := fmt.Sprintf("%s-%s.%s-template",
+				ds.Inputs[0].Streams[0].DataStream.Type,
+				pkgManifest.Name,
+				transform.Name,
+			)
+
+			// TODO: use the docs obtained here for the fields validator too
+			// In order to compare the mappings, it is required to wait until the documents has been
+			// ingested in the given transform index
+			// It looks like that not all documents ingested previously in the main data stream are going
+			// to be ingested in the destination index of the transform. That's the reason to disable
+			// the asserts
+			waitOpts := waitForDocsOptions{
+				period: 5 * time.Second,
+				// Add specific timeout to ensure transform processes the documents (depends on "delay" field?)
+				timeout: 10 * time.Minute,
+				stopCondition: func(hits *hits, oldHits int) (bool, error) {
+					deleted, err := r.getDeletedDocs(ctx, destIndexTransform)
+					if err != nil {
+						return false, err
+					}
+					foundDeleted := deleted > 0
+					// Wait at least for the number of documents found after running the transform preview
+					foundHits := hits.size() >= len(transformDocs)
+					if foundDeleted {
+						logger.Debugf("Found %d deleted docs in %s index", deleted, destIndexTransform)
+					}
+					if foundHits {
+						logger.Debugf("Found %d hits in %s index", hits.size(), destIndexTransform)
+					}
+					return foundDeleted || foundHits, nil
```

Comment:
Could it happen that there is only one deleted document, but we should wait for more hits found?

Reply:
The issue here is that there are packages that there are just deleted documents and never receive any new document. That's the reason why it is here just an "or" condition.

I would need to check if there is any package with deleted and new documents in the index :eyes:

Reply:
Found a package `ti_eset` where this scenario happens.

When validating the transform, there are 50 deleted documents in the destination index, and just one document not deleted. However, there were 26 documents returned via the preview API.

So in that package, waiting for docs would exit with error due to timeout, if we would wait for the same amount of documents found in the preview.

```
2025/01/23 13:33:26 DEBUG Found 26 documents in preview for transform "logs-ti_eset.apt_latest_ioc-default-0.3.0"
...
2025/01/23 13:39:06 DEBUG Found 50 deleted docs in logs-ti_eset_latest.dest_apt-3 index
2025/01/23 13:39:11 DEBUG found 1 hits in logs-ti_eset_latest.dest_apt-3 index or data stream
....
```

Reply:
Found a package `ti_eset` where this scenario happens.

When validating the transform, there are 50 deleted documents in the destination index, and just one document not deleted. However, there were 26 documents returned via the preview API.

So in that package, waiting for docs would exit with error due to timeout, if we would wait for the same amount of documents found in the preview.

```
2025/01/23 13:33:26 DEBUG Found 26 documents in preview for transform "logs-ti_eset.apt_latest_ioc-default-0.3.0"
...
2025/01/23 13:39:06 DEBUG Found 50 deleted docs in logs-ti_eset_latest.dest_apt-3 index
2025/01/23 13:39:11 DEBUG found 1 hits in logs-ti_eset_latest.dest_apt-3 index or data stream
....
```